internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,242 @@
+        ConfigParser.parseJIRATicketInfo();
+        String credentials = new String(getUserName()) + ":" + new String(getPassWord());
+        String auth;
+        auth = new String(Base64.encodeBase64(credentials.getBytes(UTF_8)), UTF_8);
+        for (Map.Entry<String, String> entry : responseMapper.entrySet()) {
+            if (isVulnerabilityExistInReport(entry.getValue())) {
+                String summary = entry.getKey() + ":" + ticketSubject;
+                String issueKey = checkForIssueExistence(auth, summary, projectKey);
+                if (StringUtils.isBlank(issueKey)) {
+                    createJIRAIssue(entry.getKey(), fileMapper.get(entry.getKey()), auth, summary);
+                } else {
+                    jira.invokePostComment(auth, IssueCreatorConstants.WSO2_JIRA_BASE_URL +
+                            IssueCreatorConstants.JIRA_REST_API_URL +
+                            "/" + issueKey + "/comment", createComment(entry.getKey()));
+                    log.info("[JS_SEC_DAILY_SCAN] JIRA Comment added : " + createComment(entry.getKey()));
+                    jira.invokePutMethodWithFile(auth, IssueCreatorConstants.WSO2_JIRA_BASE_URL +
+                            IssueCreatorConstants.JIRA_REST_API_URL + "/" +
+                            issueKey + "/attachments", fileMapper.get(entry.getKey()));
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Create JIRA Issue.
+     *
+     * @param product  product name
+     * @param filePath File path where the generated report is located. It will attached with ticket.
+     * @throws IssueCreatorException exception occurred while creating issue ticket.
+     */
+    private void createJIRAIssue(String product, String
+            filePath, String auth, String summary) throws IssueCreatorException {
+        String issue;
+        String assignee = null;
+        if (product.contains(JSScannerConstants.AM)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.AM);
+        } else if (product.contains(JSScannerConstants.INTEGRATION)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.INTEGRATION);
+        } else if (product.contains(JSScannerConstants.STREAMPROCESSOR)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.STREAMPROCESSOR);
+        } else if (product.contains(JSScannerConstants.IDENTITYSERVER)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.IDENTITYSERVER);
+        } else if (product.contains(JSScannerConstants.OB)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.OB);
+        }
+
+        if (assignee == null) {
+            assignee = IssueCreatorConstants.DEFAULT_ASSIGNEE;
+        }
+
+        String jiraBaseUrl = IssueCreatorConstants.WSO2_JIRA_BASE_URL;
+
+        String issueTobeCreated = createNewTicket(product, assignee, summary);
```

Comment:
Proper camel-casing need to be used. 'issueTobeCreated' should be 'issueToBeCreated'

Reply:
Resolved